pytorch_blade/torch_blade/quantization/prepare_data.py

```diff
@@ -0,0 +1,247 @@
+                      %y : Tensor,
+                      %5 : __torch__.torch.jit._script.ScriptModule):
+                    %2 : Tensor, %3 : Tensor = prim::FusionGroup_0(%x, %y)
+                    return (%2, %3)
+                with prim::FusionGroup_0 = graph(%x_ : Tensor, %y_ : Tensor):
+                    ...
+                    ...
+                    some nodes of the fusion group
+
+            After this function, the graph is like:
+                graph(%self.1 : __torch__.___torch_mangle_2.MyModel,
+                      %x : Tensor,
+                      %y : Tensor,
+                      %5 : __torch__.torch.jit._script.ScriptModule):
+                    %6 : __torch__.libtorch_executable_fusion_group_0 = prim::GetAttr[name="libtorch_executable_fusion_group_0"](%5) # noqa
+                    %7 : Tensor[] = prim::CallMethod[name="forward"](%6, %x, %y)
+                    %9 : Tensor, %10 : Tensor = prim::ListUnpack(%7)
+                    return (%9, %10)
+        """
+        get_attr_nodes_for_fusion_group = []
+
+        for idx, g_node in enumerate(group_nodes(self.graph)):
+            subgraph = g_node.g("Subgraph").copy()
+            group_name = f"{self.executable_fusion_prefix}_{idx}"
+            fallback_module = utils.subgraph_to_module(subgraph, group_name)
+            self.custom_module_owner._c._register_attribute(
+                group_name, fallback_module._type(), fallback_module
+            )
+
+            # getattr node for the fusion group
+            get_attr = self.graph.createGetAttr(self.custom_module_owner_inp, group_name)
+            self.graph.appendNode(get_attr)
+            get_attr.moveAfter(g_node)
+            get_attr.output().setType(fallback_module._type())
+            get_attr_nodes_for_fusion_group.append(get_attr)
+
+            # call the forward function of the fusion group
+            call_method = self.graph.create("prim::CallMethod")
+            self.graph.appendNode(call_method)
+            call_method.moveAfter(get_attr)
+            # prim::CallMethod can not return multiple output, so we set
+            # the type of output to Tensor List
+            call_method.output().setType(torch_blade.tools.get_list_tensor_type())
+            call_method.s_("name", "forward")
+            call_method.addInput(get_attr.output())
+            for inp in g_node.input_list():
+                call_method.addInput(inp)
+
+            # unpack the output of the forward method
+            list_unpack = self.graph.create('prim::ListUnpack')
+            self.graph.appendNode(list_unpack)
+            list_unpack.moveAfter(call_method)
+            list_unpack.addInput(call_method.output())
+            list_unpack.eraseOutput(0)
+            for out in g_node.output_list():
+                lu_out = list_unpack.addOutput()
+                lu_out.setType(out.type())
+                out.replaceAllUsesWith(lu_out)
+
+            g_node.destroy()
```

Comment:
Nit: can we reuse [torch_blade/clustering/support_group_conversion.py#L41](https://github.com/alibaba/BladeDISC/blob/main/pytorch_blade/torch_blade/clustering/support_group_conversion.py#L41)

Reply:
There might be some slight difference:
1.  The compiled engine takes List of Tensor as inputs while here we don't group the inputs into one list since the input of each `prim::FusionGroup` is not a List. 
2.  The `prim::GetAttr` should be returned.
3.  The custom module owner is the a new-added input while the `_replace_group_with_engine` uses the first original input.

Difference 2 and 3 is easy to solve by extending `_replace_group_with_engine`. However I can not found an elegant way to solve the difference 1. Any suggestions?

Reply:
done, I extended the `replace_group_with_engine` and reuse it in ` make_fusion_group_executable`